src/frontend/screens/Settings/CreateOrJoinProject/CreateProject/index.tsx

```diff
@@ -131,70 +130,51 @@ export const CreateProject: NativeNavigationComponent<'CreateProject'> = ({
       configFileResult?.type === 'success'
         ? configFileResult.file.uri
         : undefined;
+    const configPath = fileUri && convertFileUriToPosixPath(fileUri);
+
+    const onProjectCreated = () =>
+      navigation.navigate('ProjectCreated', {name: projectName});
+
+    const onError = (err: unknown) => {
+      Sentry.captureException(err);
+      navigation.navigate('ErrorBottomSheet');
+    };
 
     if (projectInfo.role === 'solo') {
-      if (fileUri) {
-        importProjectConfig.mutate(
-          {configPath: convertFileUriToPosixPath(fileUri)},
+      const updateSettings = () =>
+        updateSettingsMutation.mutate(
+          {name: projectName},
```

Comment:
so, I think this is on the line of being a style preference vs an actual problem, so im happy to hear your side if you think we should keep it this way.

But i find it a little confusing that we are creating 3 functions (`onProjectCreated`, `onError`, `updateSettings`) inside `handleCreateOrUpdateProject`. It feel more straightforward to create those function outside of `handleCreateOrUpdateProject`. What do you think?